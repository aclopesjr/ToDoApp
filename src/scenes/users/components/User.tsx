import React from 'react';
import {StyleSheet, Text, View, TouchableOpacity} from 'react-native';
import PropTypes from 'prop-types';

const User = (props: {
  name: string;
  username: string;
  email: string;
  onPress: () => void;
}) => {
  return (
    <TouchableOpacity style={styles.viewItem} onPress={props.onPress}>
      <View style={styles.thumbnail} />
      <View>
        <Text style={styles.title}>{props.name}</Text>
        <Text style={styles.subtitle}>{props.username}</Text>
        <Text style={styles.subtitle}>{props.email}</Text>
      </View>
    </TouchableOpacity>
  );
};

export default User;

User.propTypes = {
  name: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
};

const styles = StyleSheet.create({
  viewItem: {
    flex: 1,
    flexDirection: 'row',
    padding: 18,
  },
  thumbnail: {
    width: 40,
    height: 40,
    backgroundColor: '#f5f5',
    marginEnd: 12,
    marginTop: 12,
  },
  title: {
    fontSize: 20,
    color: 'black',
  },
  subtitle: {
    fontSize: 16,
    color: 'gray',
  },
});
