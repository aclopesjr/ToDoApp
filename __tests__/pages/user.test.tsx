import React from 'react';
import {shallow} from 'enzyme';
import User from '../../src/scenes/users/components/User';

describe('Snapshot User Component', () => {
  it('Renders User correctly', () => {
    const component = shallow(
      <User
        name="Name"
        username="username"
        email="email@email.com"
        onPress={() => {}}
      />,
    );
    expect(component).toMatchSnapshot();
  });
});
