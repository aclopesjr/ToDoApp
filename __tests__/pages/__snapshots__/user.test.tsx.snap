// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot User Component Renders User correctly 1`] = `ShallowWrapper {}`;
